### DISCUSSION TITLE
Vertically splitting a PDF document

### DISCUSSION THREAD

**Discussion by dummifiedme** [pymupdf/PyMuPDF#1679] [Looking for help]
I came across some scanned PDFs which contain two pages as one. 
I have a code using pymupdf which extracts highlights and images for me.

With such books which are not technically single page per page, they mess up results. So, I was wondering if there is a way to split a PDF vertically have a logical sequence in the new PDF. I can keep that snippet handy and without any need of external tools.

From what I have learnt about pymupdf, its fairly robust and can manipulate a PDF heavily. I hope this is something doable as well?

**Comment by dummifiedme** [comment]
From what I can think of (programming noob):

We can take each page and in  a loop,
- append one half of a page in a list
- append other half in the same list

Edit:Can something like this work?

I actually found the "Recipe" in the documentation! :smile: 

Edit 2: I modified the code for my use, but I notice that the annotations are lost, and I kind of understand why.

**What should we do if we wish to preserve the annotations as well?**

**Reply by JorjMcKie** [reply]
> What should we do if we wish to preserve the annotations as well?

The easiest way would be to convert annotations to regular page content. In PyMuPDF this can be achieved by doing a PDF-to-PDF conversion: `pdfbytes = doc.convert_to_pdf(...)`, then `newpdf = fitz.open("pdf", pdfbytes)`.

The new PDF should hopefully look exactly like the original one, except that annotations now have become drawings, regular text, etc.
In some cases however you may run into unsupported fonts and minor layout garblings.
Try it out.
